cmd/kube-controller-manager/app/core.go

```diff
@@ -531,3 +559,47 @@ func processCIDRs(cidrsList string) ([]*net.IPNet, bool, error) {
 
 	return cidrs, dualstack, nil
 }
+
+// setNodeCIDRMaskSizes sets the IPv4 and IPv6 node cidr mask sizes to the value provided
+// for --node-cidr-mask-size flag
+func setNodeCIDRMaskSizes(cfg nodeipamconfig.NodeIPAMControllerConfiguration, ipv4Mask, ipv6Mask *int) error {
+	// NodeCIDRMaskSizeIPv4 and NodeCIDRMaskSizeIPv6 can be used only for dual-stack clusters
+	if cfg.NodeCIDRMaskSizeIPv4 != 0 || cfg.NodeCIDRMaskSizeIPv6 != 0 {
+		return fmt.Errorf("usage of --node-cidr-mask-size-ipv4 and --node-cidr-mask-size-ipv6 are not allowed with non dual-stack clusters")
+	}
+	if cfg.NodeCIDRMaskSize != 0 {
+		*ipv4Mask = int(cfg.NodeCIDRMaskSize)
+		*ipv6Mask = int(cfg.NodeCIDRMaskSize)
+	}
+	return nil
+}
+
+// setNodeCIDRMaskSizesDualStack sets the IPv4 and IPv6 node cidr mask sizes to the value provided
+// for --node-cidr-mask-size-ipv4 and --node-cidr-mask-size-ipv6 respectively
+func setNodeCIDRMaskSizesDualStack(cfg nodeipamconfig.NodeIPAMControllerConfiguration, ipv4Mask, ipv6Mask *int) error {
```

Comment:
Curious why `*int` here? Can we return (int, int) instead and set those to `nodeCIDRMaskSizeIPv4` and `nodeCIDRMaskSizeIPv6` instead?

Reply:
changed it to return (int, int, err)